### ISSUE TITLE
Deal with non-V3 certificates better

### ISSUE THREAD

**Issue by briansmith** [briansmith/webpki#53]
Either return a different error code, or document somewhere that the most likely cause of BadDER is a non-v3 certificate.

**Comment by jonas-schievink** [comment]
Ah, but apparently the error returned when the version field is missing completely is still `BadDER`, so this is still relevant.

**Comment by briansmith** [comment]
> Ah, but apparently the error returned when the version field is missing completely is still `BadDER`, so this is still relevant.

Right. The way to encode "version 1" is by not encoding the version at all, unfortunately. The best way to fix that would be to change *ring*'s DER parser to return different errors for "the tag isn't what was expected" vs. other errors. I would take PRs to *ring* and to webpki to do that.

**Comment by briansmith** [comment]
> The best way to fix that would be to change _ring_'s DER parser to return different errors for "the tag isn't what was expected" vs. other errors. I would take PRs to _ring_ and to webpki to do that.

#90 is a very similar problem and the best solution seems to be similar to what I suggested above.
